pkg/R/serialize.R

```diff
@@ -16,10 +16,11 @@ writeObject <- function(con, object, writeType = TRUE) {
   # NOTE: In R vectors have same type as objects. So we don't support
   # passing in vectors as arrays and instead require arrays to be passed
   # as lists.
+  type <- class(object)[[1]]  # class of POSIXlt is c("POSIXlt", "POSIXt")
   if (writeType) {
-    writeType(con, class(object))
+    writeType(con, type)
   }
-  switch(class(object),
+  switch(type,
```

Comment:
can you add a type for `NULL` ? We'll then need a `readObject` on the Scala side that will parse the `NULL` and then convert it to the null type for SparkSQL.